python/src/scipp/compat/mantid.py

```diff
@@ -27,20 +27,19 @@ def convert_instrument(ws):
 
 
 def init_pos_spectrum_no(nHist, ws):
-
-    pos = np.zeros([nHist, 3])
-    num = np.zeros([nHist], dtype=np.int32)
+    pos = sc.Variable([sc.Dim.Spectrum], [nHist], unit=sc.units.m,
+                      dtype=sc.dtype.vector_3_double)
+    num = sc.Variable([sc.Dim.Spectrum], [nHist], dtype=np.float32)
 
     spec_info = ws.spectrumInfo()
     for i in range(nHist):
         if spec_info.hasDetectors(i):
             p = spec_info.position(i)
-            pos[i, :] = [p.X(), p.Y(), p.Z()]
+            pos[sc.Dim.Spectrum, i].value = (p.X(), p.Y(), p.Z())
```

Comment:
Did you verify that this does not cause a performance regression (using `Variable` instead of `numpy.ndarray` on the left-hand side)?

Reply:
The performance degrades, but anyway `numpy.array` of shape `(whatever, 3)` shouldn't be silently converted to `variable` with 3d vectors.

Reply:
Not silently, but I think we may need explict conversion. How long does this take, e.g., for 1e6 pixels?